enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,411 @@
+
+###### `--manifests`
+
+Generate a package containing a CSV and owned CRD manifests, and updates the
+[package manifest file][package-manifest-doc]. This step uses a base generated
+with `--kustomize`, or creates a new one. This subcommand is interactive if a
+base does not exist.
+
+New project layout defaults:
+* Output directory: `config/packages/<version>`
+* Read from: stdin, `--apis-dir="api(s)"`
+
+Old project layout defaults:
+* Output directory: `deploy/olm-catalog/<operator-name>/<version>`
+* Read from: `--manifest-root="deploy"`, `--crds-dir="deploy/crds"`, `--apis-dir="pkg/apis"`
+
+Notable flags: `--version=<semver>`, `--channel=<channel-name>`, `--default-channel=<channel-name>`
+
+#### `make` recipes (new project layouts)
+
+New project layouts are scaffolded with a [Makefile][kb-makefile] containing a
+set of pre-defined recipes for generating manifests, building images, etc. To align
+with this idiom, recipes should be amended/added to include `generate bundle` by
+default. If a user does not want these features turned on in their project, they
+can manually remove the amendments/additions.
+
+###### Modifying the base scaffold
+
+`operator-sdk init` will scaffold a Makefile containing the following amendments/additions:
+
+```make
+# Current Operator version
+VERSION ?= 0.0.1
+# Default bundle image tag
+BUNDLE_IMG ?= controller-bundle:$(VERSION)
+# Options for 'bundle-build'
+ifeq ($(origin BUNDLE_CHANNELS), undefined)
+BUNDLE_CHANNELS = --channels=$(BUNDLE_CHANNELS)
+endif
+ifeq ($(origin BUNDLE_DEFAULT_CHANNEL), undefined)
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(BUNDLE_DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+...
+
+# Existing manifests recipe, append 'generate bundle'
+manifests:
+  ...
+  operator-sdk generate bundle --kustomize
+
+...
+
+# Generate bundle manifests
+bundle: manifests
+	kustomize build config/bundle | operator-sdk generate bundle --manifests --version $(VERSION)
+
+# Generate bundle manifests with a particular version and metadata,
+# validate generated bundle files, and build the bundle image.
+bundle-build: manifests
+	kustomize build config/bundle | operator-sdk generate bundle --manifests --metadata --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
+	operator-sdk bundle validate config/bundle
+	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+```
+
+A user can either invoke `make bundle` or `make bundle-build` or both; there is
+no precedence requirement.
+
+[opm-bundle-generate]:https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#generate-bundle-annotations-and-dockerfile
+[package-manifest-doc]:https://github.com/operator-framework/operator-lifecycle-manager#discovery-catalogs-and-automated-upgrades
+[kb-makefile]:https://github.com/kubernetes-sigs/kubebuilder/blob/7bdcf98/pkg/scaffold/internal/templates/makefile.go#L58-L151
+
+## Design Details
+
```

Comment:
The design here is defining that the bundle targets will be added in the Makefile when the project is created. However, it does not address the story: 

> I am as an Operator Developer would like to use the SDK tool to integrate a project built with Kubebuiler with OLM

**My suggestion is:** 
SDK provides a command that will ADD the files/makefile target required to support it. E.g. 

- I run `operator-sdk add bundle` then, the scaffold project is updated to support this feature. 
- If the project is built with SDK then this support is added by default which means that when we run `operator-sdk 
 init` it also will call `operator-sdk add bundle` for example.

Reply:
https://github.com/operator-framework/enhancements/pull/16#discussion_r431441168